components/tikv_util/src/sys/disk.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+use fail::fail_point;
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+pub static DISK_FULL: AtomicBool = AtomicBool::new(false);
+pub static DISK_RESERVED: AtomicU64 = AtomicU64::new(0);
+
+pub fn set_disk_full() {
+    DISK_FULL.store(true, Ordering::Release);
+}
+
+pub fn clear_disk_full() {
+    DISK_FULL.store(false, Ordering::Release);
+}
+
+pub fn is_disk_full() -> bool {
+    DISK_FULL.load(Ordering::Acquire)
+}
+
+//underscore to disable unused warning
+pub fn follower_disk_full_precheck(_store_id: u64) -> bool {
+    if cfg!(feature = "failpoints") {
+        fail_point!("disk_full_follower_1", _store_id == 1, |_| {
+            return true;
+        });
+        return false;
+    } else {
+        return false;
+    }
+}
+
+pub fn leader_disk_full_precheck() -> bool {
+    if cfg!(feature = "failpoints") {
+        fail_point!("disk_full_leader", |_| {
+            return true;
+        });
+        return false;
+    } else {
+        return false;
+    }
+}
+
+pub fn set_disk_reserved(v: u64) {
+    let mut value = v;
+    if v == 0 {
+        value = 5 * 1024 * 1024 * 1024;
```

Comment:
It's better to write the logic outside. For example, `Config::validate`. And I believe the logic exists already.